pdfbox/src/main/java/org/apache/pdfbox/pdmodel/graphics/state/PDGraphicsState.java

```diff
@@ -39,7 +44,8 @@
 public class PDGraphicsState implements Cloneable
 {
     private boolean isClippingPathDirty;
-    private Area clippingPath;
+    private List<GeneralPath> clippingPaths = new ArrayList<>();
+    private Map<GeneralPath, Area> clippingCache = new IdentityHashMap<>();
```

Comment:
I'm not sure is it too naive to have an unbound cache here, but it could easily be improved/size-limited.